grades/tasks.py

```diff
@@ -22,6 +28,57 @@
 
 
 @app.task
+def generate_course_certificates_for_fa_students():
+    """
+    Task that finds users with a final grade that should have a certificate and creates
+    the certificate.
+    """
+    # Fetch FinalGrades in FA programs with no associated MicromastersCourseCertificate
+    final_grade_values = (
+        FinalGrade.objects
+        .filter(
+            course_run__course__program__financial_aid_availability=True,
+            passed=True,
+        )
+        .annotate(certificate_count=Count('micromasterscoursecertificate'))
+        .filter(certificate_count=0)
+        .annotate(exam_run_count=Count('course_run__course__exam_runs'))
+        .values(
+            'id',
+            'user__id',
+            'user__username',
+            'course_run__course__id',
+            'course_run__edx_course_key',
+            'exam_run_count',
+        )
+    )
+    # Filter the results for FinalGrades that should have a MicromastersCourseCertificate created.
+    # A FinalGrade should have a certificate if:
+    #   1. No exams were scheduled for the course, or
+    #   2. The course has a scheduled exam and the user has a passing ProctoredExamGrade in the course.
+    final_grades_to_certify = filter(
+        lambda final_grade_dict: (
+            final_grade_dict['exam_run_count'] == 0 or
+            ProctoredExamGrade.objects.filter(
+                user_id=final_grade_dict['user__id'],
```

Comment:
`final_grade_dict['exam_run_count'] == 0` means that the course has no exam runs, which means that we can generate a certificate based on the `FinalGrade` only. We don't need to check for exam grades since there are no exams